src/controllers/auth.js

```diff
@@ -187,4 +193,79 @@ const registerUser = async (req, res) => {
   }
 };
 
-module.exports = { createDefaultTables, updateUser, registerUser };
+const obtainAccessToken = async (req, res) => {
+  // extract payload
+  const { username, password } = req.body.fields;
+
+  try {
+    // check if the username exists in the Db
+    let users = rowService.get({
+      tableName: '_users',
+      whereString: 'WHERE username=?',
+      whereStringValues: [username],
+    });
+
+    if (users.length <= 0) {
+      return res.status(401).send({ message: 'Invalid username or password' });
+    }
+
+    // check if the password is valid
+    let user = users[0];
+    const isMatch = await comparePasswords(password, user.hashed_password);
+
+    if (!isMatch) {
+      return res.status(401).send({ message: 'Invalid username or password' });
+    }
+
+    // get the users role from the DB
```

Comment:
for the next line, we will actually receive  an array of matching records.  I _think_ it should be named as plural (`userRoles`), then we check there is exactly 1 record, then we store the [0] element as `const userRole`.